kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/batch/BatchTableWriter.java

```diff
@@ -0,0 +1,133 @@
+
+    private final BlobInfo blobInfo;
+    private final Storage storage;
+    private final String sourceUri;
+
+    private final BigQuery bigQuery;
+    private final LoadJobConfiguration loadJobConfiguration;
+
+    private final List<Map<String, Object>> records;
+
+    /**
+     * @param bucketName The name of the bucket to which a blob containing record information should be uploaded
+     * @param blobName Full path within the bucket to the blob (without the extension) (Doesn't need to be pre-created)
+     * @param storage GCS Storage
+     * @param tableId {@link TableId} of the BigQuery table to upload to
+     * @param bigQuery {@link BigQuery} Object used to perform upload
+     * @param records Records to upload to BigQuery through GCS
+     */
+    public BatchTableWriter(String bucketName, String blobName, Storage storage,
+                            TableId tableId, BigQuery bigQuery,
+                            List<Map<String, Object>> records) {
+        BlobId blobId = BlobId.of(bucketName, blobName+".json");
+        blobInfo = BlobInfo.newBuilder(blobId).setContentType("text/json").build();
+        this.storage = storage;
+        this.sourceUri = String.format("gs://%s/%s.json", bucketName, blobName);
+
+        this.bigQuery = bigQuery;
+
+        // Check if the table specified exists
+        try {
+            if (!bigQuery.getTable(tableId).exists()) {
+                throw new BigQueryException(NOT_FOUND_ERROR_CODE, "");
+            }
+        } catch (BigQueryException | NullPointerException exception) {
+            throw new BigQueryException(NOT_FOUND_ERROR_CODE,
+                    exception + String.format("Table id for table %s does not exist", tableId.getTable()));
+        }
+
+        this.loadJobConfiguration =
+                LoadJobConfiguration.builder(tableId, sourceUri)
+                .setFormatOptions(FormatOptions.json())
+                .setCreateDisposition(JobInfo.CreateDisposition.CREATE_NEVER)
+                .build();
+
+        this.records = records;
+    }
+
+    @Override
+    public void run() {
+        //todo implement
+    }
+
+    private void transferBlobToBigQuery() throws RuntimeException {
+        Job loadJob = bigQuery.create(JobInfo.of(loadJobConfiguration));
+        String exceptionMessage = String.format("%s.\nSource URI = \"%s\"\nTable = \"%s\"",
+                "Transfer from GCS blob to BigQuery unsuccessful.",
+                loadJobConfiguration.getSourceUris(),
+                loadJobConfiguration.getDestinationTable());
+        try {
+            loadJob = loadJob.waitFor();
+        } catch (InterruptedException | TimeoutException exception) {
+            throw new RuntimeException(exceptionMessage, exception);
+        }
+    }
+
+    public Blob uploadRecords(List<Map<String, Object>> records) {
+        return uploadBlob(new ByteArrayInputStream(toJson(records).getBytes()));
+    }
+
+    private Blob uploadBlob(InputStream jsonRecords) {
```

Comment:
I feel like there should be more clarity on which functions are involved in the local -> GCS upload and which are involve in the GCS -> BQ trigger, either in method naming or documentation (or both)